src/main/scala/org/scalameter/Gen.scala

```diff
@@ -74,27 +76,27 @@ object Gen {
 
   def unit(axisName: String): Gen[Unit] = new Gen[Unit] {
     def warmupset = Iterator.single(())
-    def dataset = Iterator.single(Parameters((axisName, ())))
+    def dataset = Iterator.single(Parameters((Parameter[Unit](axisName), ())))
     def generate(params: Parameters) = params[Unit](axisName)
   }
 
-  def single[T](axisName: String)(v: T): Gen[T] = enumeration(axisName)(v)
+  def single[T: Pickler](axisName: String)(v: T): Gen[T] = enumeration(axisName)(v)
 
   def range(axisName: String)(from: Int, upto: Int, hop: Int): Gen[Int] = new Gen[Int] {
     def warmupset = Iterator.single(upto)
-    def dataset = (from to upto by hop).iterator.map(x => Parameters(axisName -> x))
+    def dataset = (from to upto by hop).iterator.map(x => Parameters(Parameter[Int](axisName) -> x))
     def generate(params: Parameters) = params[Int](axisName)
   }
 
-  def enumeration[T](axisName: String)(xs: T*): Gen[T] = new Gen[T] {
+  def enumeration[T: Pickler](axisName: String)(xs: T*): Gen[T] = new Gen[T] {
```

Comment:
Besides the `SerializationPersistor`, the benchmark classes are serialized when sent to another JVM for execution.
This also means that the generators are serialized, so we would have to avoid embedding the picklers themselves into the generator, and instead embed their classnames.